pkg/pillar/agentlog/loglevel.go

```diff
@@ -69,98 +75,101 @@ func getRemoteLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetRemoteLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
+	gc := cast.CastConfigItemValueMap(m)
 	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.RemoteLogLevel != "" {
-		return as.RemoteLogLevel, true
-	}
-	// Do we have a default value?
-	if allowDefault && gc.DefaultRemoteLogLevel != "" {
-		return gc.DefaultRemoteLogLevel, true
+	if _, ok1 := gc.AgentSettings[agentName]; ok1 {
+		as, ok := gc.AgentSettings[agentName]["remote.loglevel"]
+		if ok && as.StringValue != "" {
+			return as.Key, true
+		} else if ok && as.StringValue != "" && allowDefault {
+			// Return default remote.loglevel (global)
+			return "global", true
+		} else {
+			log.Infof("Failed to find loglevel settings for %s", agentName)
+		}
+	} else {
+		log.Infof("Failed to find per agent settings for %s", agentName)
```

Comment:
Similar comments as above..

For remote.loglevel - the hierarchy is:
   if agentSetting found {
       if agentSettings["remote.loglevel"] found {
             remoteLogLevel = agentSettings["remote.loglevel"]
        }
    }
    if remoteLogLevel == "" {
        remoteLogLevel = Global Remote Loglevel
    }